lib/ansible/module_utils/common/validator.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 Ansible Project
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+from ansible.module_utils.common.parameters import (
+    get_unsupported_parameters,
+    validate_argument_types,  # Rename this because it actually does coercion?
+    validate_argument_values,
+)
+
+from ansible.module_utils.common.text.converters import to_native
+
+
+from ansible.module_utils.common.validation import (
+    check_required_arguments,
+)
+
+
+class Validator():
+    """Argument spec validator class"""
+
+    def __init__(self):
+        self._options_context = None
+
+        self.argument_spec = {}
+        self.module_parameters = {}
+        self.error_messages = []
+        self.validated_parameters = {}
+
+    def validate(self, arg_spec, module_parameters):
+        """Validate module parameters against argument spec.
+
+        :Example:
+
+        validator = Validator()
+        validator.validate()
+        validated_parameters = validator.validated_parameters
+        errors = validator.error_messages
```

Comment:
Let's provide accessor methods for these attributes instead and consider the attributes private (maybe rename them with a leading underscore to indicate they _should_ be private). This will let us change how we store this data later on, if we choose to.

Reply:
That's a good idea. I will do that.

Reply:
Do you have an opinion on propries vs actual methods?

```python
v.get_error_messages()

# or

v.error_messages
```

And the properites would look like:
```python
@property
def error_messages(self):
    return self._error_messages

@property
def validated_parameters(self):
    return self._validated_parameters
```

Not doing much currently, but gives us the ability to intercept these calls and change the behavior in the future.

Reply:
Use of `@property` is fine, IMO. Serves the same purpose here.